examples/basic/main.go

```diff
@@ -26,7 +19,7 @@ func main() {
 	client := plugin.NewClient(&plugin.ClientConfig{
 		HandshakeConfig: handshakeConfig,
 		Plugins:         pluginMap,
```

Comment:
I guess this is a more general usage question. Do we really need these configs, especially `pluginMap` here? That means it needs to know what plugins are available at compile time right? Kinda destroy the whole purpose of having plugins